src/pages/trainings/checkout/[reference].tsx

```diff
@@ -39,14 +36,44 @@ const CheckoutPage: NextPage<{
 
 	const isxpired = new Date(trainingOrder.expiredAt).getTime() < Date.now();
 
+	const updateTrainingOrderData = async () => {
+		await fetch('/api/update-training-order', {
+			method: 'POST',
+			headers: {
+				'Content-Type': 'application/json',
+			},
+			body: JSON.stringify({ reference: trainingOrder.reference }),
+		})
+			.then(res => res.json())
+			.then(data => {
+				setIsSuccessful(true);
+				setPaidOrder(data);
+			})
+			.catch(err => {
+				console.log(err);
+				setIsSuccessful(false);
+			});
+	};
+	const updateTrainingData = async () => {
+		await fetch('/api/update-training-bookedspot', {
+			method: 'POST',
+			headers: {
+				'Content-Type': 'application/json',
+			},
+			body: JSON.stringify({ id: trainingOrder.trainingId }),
+		}).catch(err => {
+			console.log(err);
+		});
+	};
+
 	/**
 	 *
 	 * @param reference - transaction reference
 	 * @returns void
 	 * @description - this function is called when the transaction is successful
 	 *
 	 */
-	const onSuccess = (reference: any) => {
+	const onSuccess = (reference: string) => {
```

Comment:
the reference here is not a string, it's an object that has a reference attribute amongst other things

Reply:
reverted back to type any